### PR TITLE
Add Links page object and test-suit for it

### PR DESCRIPTION
**Pull request by rybignat**
add links.page.ts
update README
links.spec.ts (9 cases):
1. Verify "Home" Link Functionality
2. Verify Dynamic Link Functionality
3. Verify "Created" Link Functionality
4. Verify "No Content" Link Functionality
5. Verify "Moved" Link Functionality
6. Verify "Bad Request" Link Functionality
7. Verify "Unauthorized" Link Functionality
8. Verify "Forbidden" Link Functionality
9. Verify "Not Found" Link Functionality

### CODE REVIEW

File: pageObjects/links.page.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  constructor (page: Page) {
+    this.page = page
+    this.homeLink = page.locator('//p//a[@id="simpleLink"]')
+    this.homeDynamicLink = page.locator('//p//a[@id="dynamicLink"]')
+    this.textMessageArea = page.locator('//p[@id="linkResponse"]')
+  }
+
+  async clickLinkByName (key: string): Promise<void> {
```
Comment (AliaksandrPuchok) [inline pageObjects/links.page.ts]: clickLinkByName, но в аргументах key:

File: pageObjects/links.page.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  async checkResultMessage (statusCode: number, statusText: string): Promise<void> {
+    if (statusText === 'Moved') {
+      statusText = 'Moved Permanently'
+    }
+    const result: string = await this.textMessageArea.textContent() as string
+    const expectedResult = `Link has responded with staus ${statusCode} and status text ${statusText}`
+    expect(result).toEqual(expectedResult)
+  }
```
Comment (AliaksandrPuchok) [inline pageObjects/links.page.ts]: Как тебе такой вариант?

  ```
async checkResultMessage(statusCode: number, statusText: string): Promise<void> {
    const finalStatusText = statusText === 'Moved' ? `${statusText} Permanently` : statusText;
  
    const result = await this.textMessageArea.textContent()
    const expectedResult = `Link has responded with staus ${statusCode} and status text ${finalStatusText}`;
  
    expect(result).toEqual(expectedResult);
  }
```

File: pageObjects/links.page.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  async checkStatusTextAndCodeInResponse (statusCode: number, statusText: string): Promise<void> {
+    if (statusText === 'Not Found') {
+      statusText = 'invalid-url'
+    }
+    await this.page.waitForResponse(response =>
+      response.url() === `https://demoqa.com/${statusText.toLowerCase().replace(' ', '-')}` && response.status() === statusCode &&
+      response.request().method() === 'GET')
```
Comment (AliaksandrPuchok) [inline pageObjects/links.page.ts]: и тут
``` 
 async checkStatusTextAndCodeInResponse(statusCode: number, statusText: string): Promise<void> {
    const finalStatusText = statusText === 'Not Found' ? 'invalid-url' :
     statusText.toLowerCase().replace(' ', '-');
  
    await this.page.waitForResponse(response => 
      response.url() === `https://demoqa.com/${finalStatusText}` &&
      response.status() === statusCode &&
      response.request().method() === 'GET'
    );
  }
  ```

File: pageObjects/links.page.ts (hunk 1 above)
Comment (rybignat) [reply pageObjects/links.page.ts]: Исправил

File: pageObjects/links.page.ts (hunk 2 above)
Comment (rybignat) [reply pageObjects/links.page.ts]: Добавил

File: pageObjects/links.page.ts (hunk 3 above)
Comment (rybignat) [reply pageObjects/links.page.ts]: Добавил

File: tests/links.spec.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  })
+
+  test('CASE_1: Verify "Home" Link Functionality', async ({ context }) => {
+    await test.step('Click on the "Home" link.(Expected result: A new tab opens, navigating to the home page', async () => {
+      await linksPage.openAndCheckNewPage(context, false)
+    })
+  })
+
+  test('CASE_2: Verify Invalid(Dynamic) Link Functionality ("Home9IiLG")', async ({ context }) => {
+    await test.step('Click on the "Home9IiLG" link.(Expected result: A new tab opens, navigating to the home page', async () => {
```
Comment (AliaksandrPuchok) [inline tests/links.spec.ts]: Invalid?
test title ("Home9IiLG")
test.step ("Home9IiLG")

Comment (rybignat) [reply tests/links.spec.ts]: Исправил

**Review by AliaksandrPuchok — APPROVED** [review]

**Pull request merged**